src/insights_client/__init__.py

```diff
@@ -214,16 +204,26 @@ def run_phase(phase, client, validated_eggs):
     ] + sys.argv[1:]
     config = client.get_conf()
 
-    all_eggs = [ENV_EGG, NEW_EGG] + validated_eggs
+    all_eggs = [NEW_EGG] + validated_eggs
+    if ENV_EGG is not None:
+        all_eggs = [ENV_EGG] + all_eggs
+        logger.debug("ENV_EGG defined as %s." % ENV_EGG)
+    else:
+        logger.debug("ENV_EGG not defined, skipping.")
+
+    for egg in all_eggs:
+        if config["gpg"]:
+            if not os.path.isfile(egg):
+                logger.debug("%s is not a file, can't GPG verify. Skipping." % (egg,))
+                continue
+            logger.debug("Verifying %s..." % egg)
```

Comment:
We shouldn't need this. We get the egg information in both the happy and sad path.
- If the validation fails, it's logged on line 221.
- If it succeeds, we'll get the log on line 226.

Reply:
Although logging before and after may look redundant, I see the advantage of having a pre-message like this that in case of 💥, it is the last thing printed. A failure message may not be logged in case the 💥 is too bad so that it’s not rescued from gracefully. 

The success message is probably more useless, but it nicely pairs with the pre-message. Especially for heavy operations like spawning a subprocess.